homeassistant/components/recorder/__init__.py

```diff
@@ -405,13 +410,33 @@ def async_purge(now):
                     # Must catch the exception to prevent the loop from collapsing
                     _LOGGER.exception("Error adding state change: %s", err)
 
+                if "new_state" in event.data:
+                    self._old_state_ids[dbstate.entity_id] = dbstate.state_id
+                elif dbstate.entity_id in self._old_state_ids:
+                    del self._old_state_ids[dbstate.entity_id]
+
             # If they do not have a commit interval
             # than we commit right away
             if not self.commit_interval:
                 self._commit_event_session_or_retry()
 
             self.queue.task_done()
 
+    def _get_old_state_id(self, entity_id):
+        """Find the last state_id for an entity_id in memory cache or the database."""
+        if entity_id in self._old_state_ids:
+            return self._old_state_ids[entity_id]
+        query = (
+            self.event_session.query(States.state_id)
```

Comment:
I wonder if we even should do this. We don't want to connect state changes between restarts of Home Assistant, so maybe we can just always just rely on the dict ?

Reply:
Currently we exclude entities from being displayed in the logbook when there is no `old_state`.  If we didn't link the `old_state` all the new states right after startup would not show up in the logbook.  

We could

1. Remove the lookup and prevent these from showing up in logbook
2. Set the `old_state_id` to `0` on startup to indicate restarts
3. something else?

Reply:
Right now none of the new states after startup are showing up in the logbook. So that would be the same behavior?

Reply:
After thinking about it for a while, it seems like the state change event right after restart is noise. I think it should come out as you suggest.  

I'm removing it now